src/secp256k1/src/cpp/Scalar.cpp

```diff
@@ -228,7 +245,8 @@ Scalar Scalar::hash(const unsigned char* data, size_t len) {
      throw "Scalar: hashing overflowed";
     }
     Scalar result_(&result);
-    result_.mod_p();
+    if(mod)
+        result_.mod_p();
```

Comment:
why should not always mod?

Reply:
As far as I could tell, it's inn-accurate to immediately take a hash outside of the range (0...p-1). So we call the hash function until we get a result within the correct range. Maybe this is unnecessary though. @jivanyan any input?

Reply:
Why we call serialize() and generate() in memberFromSeed? Actually we do the same thing 2 times.
I think we just need to call has() and always do mod_p() here.